doc/tutorials/dnn/dnn_yolo/dnn_yolo.markdown

```diff
@@ -51,4 +51,151 @@ $ example_dnn_object_detection --config=[PATH-TO-DARKNET]/cfg/yolo.cfg --model=[
+                                --classes=<path_to_class_names_file> \
+                                --thr=<confidence_threshold> \
+                                --nms=<non_maximum_suppression_threshold> \
+                                --mean=<mean_normalization_value> \
+                                --scale=<scale_factor> \
+                                --yolo=<yolo_model_version> \
+                                --padvalue=<padding_value> \
+                                --paddingmode=<padding_mode> \
+                                --backend=<computation_backend> \
+                                --target=<target_computation_device>
+@endcode
+
+- --input: File path to your input image or video. If omitted, it will capture frames from a camera.
+- --classes: File path to a text file containing class names for object detection.
+- --thr: Confidence threshold for detection (e.g., 0.5).
+- --nms: Non-maximum suppression threshold (e.g., 0.4).
+- --mean: Mean normalization value (e.g., 0.0 for no mean normalization).
+- --scale: Scale factor for input normalization (e.g., 1.0).
+- --yolo: YOLO model version (e.g., YOLOv3, YOLOv4, etc.).
+- --padvalue: Padding value used in preprocessing (e.g., 114.0).
+- --paddingmode: Method for handling image resizing and padding. Options: 0 (resize without extra processing), 1 (crop after resize), 2 (resize with aspect ratio preservation).
+- --backend: Selection of computation backend (0 for automatic, 1 for Halide, 2 for OpenVINO, etc.).
+- --target: Selection of target computation device (0 for CPU, 1 for OpenCL, etc.).
+- --device: Camera device number (0 for default camera). If --input is not provided camera with index 0 will used by default.
+
+Here `mean`, `scale`, `padvalue`, `paddingmode` should exactly match those that we dicussed in preprocessing section in oder for the model to match result in PyTorch
+
+### Building a Custom Pipeline
+
+Sometimes there is a need to make some custom adjustments in the inference pipeline. With OpenCV's framework this is also quite easy to achieve. Below we will outline simple starter pipeline that does the same thing as CLI cammand. You can adjust it according to your needs.
+
+- Import required libraries
+
+@snippet samples/dnn/yolo_detector.cpp includes
+
+- Read ONNX graph and create neural network model.
+
+@snippet samples/dnn/yolo_detector.cpp read_net
+
+- Read image and preprocess it
+
+@snippet samples/dnn/yolo_detector.cpp preprocess_params
+@snippet samples/dnn/yolo_detector.cpp preprocess_call
+@snippet samples/dnn/yolo_detector.cpp preprocess_call_func
+
+- Inference
+
+@snippet samples/dnn/yolo_detector.cpp forward_buffers
+@snippet samples/dnn/yolo_detector.cpp forward
+
+
+- Post-Processessing
+
+For post-procssing of the model output we will need to use function [`yoloPostProcess`](https://github.com/opencv/opencv/blob/ef8a5eb6207925e8f3055a82e90dbd9b8d10f3e3/modules/dnn/test/test_onnx_importer.cpp#L2650).
```

Comment:
let's add some description of Yolo output: general logic, output shape, semantic. Some notes on exceptions like Yolo v8.

Reply:
There is information about output is added in the section #preprocessing and output